contracts/modules/Checkpoint/Voting/APLCR/AdvancedPLCRVotingCheckpoint.sol

```diff
@@ -530,22 +526,53 @@ contract AdvancedPLCRVotingCheckpoint is AdvancedPLCRVotingCheckpointStorage, Vo
      */
     function changeBallotExemptedVotersListMulti(uint256 _ballotId, address[] calldata _exemptedAddresses, bool[] calldata _exempts) external withPerm(ADMIN) {
         _isValidLength(_exemptedAddresses.length, _exempts.length);
-        // Check for the ballots array out of bound
-        _checkIndexOutOfBound(_ballotId);
-        // Check the valid stage of ballot
-        require(getCurrentBallotStage(_ballotId) == Stage.COMMIT || getCurrentBallotStage(_ballotId) == Stage.PREP);
         for (uint256 i = 0; i < _exemptedAddresses.length; i++) {
             _changeBallotExemptedVotersList(_ballotId, _exemptedAddresses[i], _exempts[i]);
         }
     }
 
     function _changeBallotExemptedVotersList(uint256 _ballotId, address _exemptedAddress, bool _exempt) internal {
+         // Check for the ballots array out of bound
+        _checkIndexOutOfBound(_ballotId);
+        // Check the valid stage of ballot
+        require(getCurrentBallotStage(_ballotId) == Stage.PREP);
         require(_exemptedAddress != address(0), "Invalid address");
         require(ballots[_ballotId].exemptedVoters[_exemptedAddress] != _exempt, "No change");
         ballots[_ballotId].exemptedVoters[_exemptedAddress] = _exempt;
         emit ChangedBallotExemptedVotersList(_ballotId, _exemptedAddress, _exempt);
     }
 
+    /**
+     * Change the global exempted voters list
+     * @param _voter Address of the voter
+     * @param _exempt Whether it is exempted or not
+     */
+    function changeDefaultExemptedVotersList(address _voter, bool _exempt) public {
+        require(!_isAnyBallotRunning());
+        super.changeDefaultExemptedVotersList(_voter, _exempt);
+    }
+
+    /**
+     * Change the global exempted voters list
+     * @param _voters Address of the voter
+     * @param _exempts Whether it is exempted or not
+     */
+    function changeDefaultExemptedVotersListMulti(address[] memory _voters, bool[] memory _exempts) public {
+        require(!_isAnyBallotRunning());
+        super.changeDefaultExemptedVotersListMulti(_voters, _exempts);
+    }
+
+    function _isAnyBallotRunning() internal view returns (bool isAnyBallotActive) {
+        uint256 length = ballots.length;
+        isAnyBallotActive = false;
+        for (uint256 i = 0; i < length; i++) {
```

Comment:
@satyamakgec this is a bit scary with an unbounded loop. Is it possible to keep a counter (in global state) of active ballots instead of looping?

Reply:
There is no other way to do it until unless we are comfortable to allow issuer manually end the ballot. IMO this will not affect too much first it is a least callable function.

I reversed the loop as the biggest ballotId has more chances to be in the active stage, this will not completely remove the threat but will lower it.